### PR TITLE
fix(libs): 3rd party library install fixes

### PR DESCRIPTION
**Pull request by jkuri**
Closes https://github.com/angular/angular-cli/issues/152 and various other issues installing and injecting 3rd party libs.

### CODE REVIEW

**Comment by jkuri** [comment]
Btw, thanks to @Brocco for helping me with new prompt texts.

File: addon/ng2/commands/install.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ module.exports = Command.extend({
   run: function (commandOptions, rawArgs) {
     if (!rawArgs.length) {
       var msg = 'The `ng install` command must take an argument with ' +
-        'at least one package name.';
+        'package name.';
```
Comment (cironunes) [inline addon/ng2/commands/install.js]: "with a/the package name"

missing "a/the"

Comment (jkuri) [inline addon/ng2/commands/install.js]: Thanks for noticing @cironunes, I fixed it.

**Comment by filipesilva** [comment]
It looks like there's a lot less tests now. What is the reasoning? 
Just curious really.

**Comment by jkuri** [comment]
Before it was default that `ng install lib-name --auto-injection` injected everything that was exported the library in bootstrap script automatically. That wasn't okay, because we (if) want only providers to be injected in bootstrap script. I was testing if providers, directives, pipes were injected in a separate tests.
It would be possible to test if a directives, pipes, components were injected in auto generated component, if I know how to automatically go through prompts. 
Btw, now this task goes in other way around, first injecting what and where user choose and then at the end prompts if he want providers injected in bootstrap script (if any providers in library). I think thats more reasonable and make this command really useful.

**Comment by filipesilva** [comment]
Got it. Lgtm then!

File: addon/ng2/tasks/lib-install.js [hunk 2]
```diff
@@ -46,14 +45,24 @@ module.exports = Task.extend({
...
+        return this.installProcedure()
           .then(function(resp) {
             return this.announceOKCompletion();
           }.bind(this));
       }.bind(this));
   },
 
-  installProcedure: function(autoInjection) {
+  installProcedure: function() {
     var allPackages         = {};
```
Comment (cironunes) [inline addon/ng2/tasks/lib-install.js]: minor: you can improve it to:
```js
var allPackages = { toInstall: [], toUninstall: [] };
```

Comment (jkuri) [inline addon/ng2/tasks/lib-install.js]: Improved. Thanks for analyzing the code Ciro, if you find anything else please just add more notes.

File: addon/ng2/tasks/lib-install.js [hunk 3]
```diff
@@ -46,17 +45,25 @@ module.exports = Task.extend({
           return this.announceOKCompletion();
         }
 
-        return this.installProcedure(this.autoInjection)
+        if (this.autoInjection) {
+          if (existsSync(path.resolve(process.cwd(), 'src', 'app.ts'))) {
+            var entryPoint = path.resolve(process.cwd(), 'src', 'app.ts');
```
Comment (cironunes) [inline addon/ng2/tasks/lib-install.js]: Can move this line to the top and use `existSync` on it on line 49?

File: addon/ng2/tasks/lib-install.js [hunk 4]
```diff
@@ -46,17 +45,25 @@ module.exports = Task.extend({
...
+            var packageData = this.parseFile(this.packages[0]);
```
Comment (cironunes) [inline addon/ng2/tasks/lib-install.js]: Can save `this.packages[0]` in variable and reuse it?

File: addon/ng2/tasks/lib-install.js [hunk 5]
```diff
@@ -76,16 +83,16 @@ module.exports = Task.extend({
 
           return npm('uninstall', allPackages.toUninstall, this.npmOptions, this.npm)
             .then(function() {
-              return this.processPackages(allPackages.toProcess, autoInjection)
+              return this.processPackages(allPackages.toProcess)
                 .then(function(resp) {
-                  resolve(resp);
+                  return resolve(resp);
```
Comment (cironunes) [inline addon/ng2/tasks/lib-install.js]: :+1:

**Comment by cironunes** [comment]
Just added a few minor style improvements. Otherwise LGTM

**Comment by jkuri** [comment]
Fixed. Thanks.

**Pull request closed without merging**